django/core/mail/backends/smtp.py

```diff
@@ -18,37 +18,25 @@ class EmailBackend(BaseEmailBackend):
 
     def __init__(
         self,
-        host=None,
-        port=None,
-        username=None,
-        password=None,
-        use_tls=None,
         fail_silently=False,
-        use_ssl=None,
-        timeout=None,
-        ssl_keyfile=None,
-        ssl_certfile=None,
-        **kwargs,
+        use_localtime=False,
+        **options,
     ):
-        super().__init__(fail_silently=fail_silently)
-        self.host = host or settings.EMAIL_HOST
-        self.port = port or settings.EMAIL_PORT
-        self.username = settings.EMAIL_HOST_USER if username is None else username
-        self.password = settings.EMAIL_HOST_PASSWORD if password is None else password
-        self.use_tls = settings.EMAIL_USE_TLS if use_tls is None else use_tls
-        self.use_ssl = settings.EMAIL_USE_SSL if use_ssl is None else use_ssl
-        self.timeout = settings.EMAIL_TIMEOUT if timeout is None else timeout
-        self.ssl_keyfile = (
-            settings.EMAIL_SSL_KEYFILE if ssl_keyfile is None else ssl_keyfile
-        )
-        self.ssl_certfile = (
-            settings.EMAIL_SSL_CERTFILE if ssl_certfile is None else ssl_certfile
-        )
+        super().__init__(fail_silently=fail_silently, use_localtime=use_localtime)
+        self.host = options.pop("host", None)
+        self.port = options.pop("port", None)
+        self.username = options.pop("username", None)
+        self.password = options.pop("password", None)
+        self.use_tls = options.pop("use_tls", None)
+        self.use_ssl = options.pop("use_ssl", None)
+        self.ssl_keyfile = options.pop("ssl_keyfile", None)
+        self.ssl_certfile = options.pop("ssl_certfile", None)
```

Comment:
I may be missing something, but don't you still need to pull in the deprecated settings values here if this is *not* being initialized from an EMAIL_PROVIDERS OPTIONS dict?

(Otherwise that logic has to go in mail.get_connection(), as a special case for the SMTP EmailBackend, which seems complicated.)

Also, I think this needs to know whether it's being initialized from a provider or not. If it's from a provider, OPTIONS dict, it shouldn't use *any* deprecated settings. If it's *not* from a provider, deprecated settings are OK if not provided as args. (But you don't want to mix the two.) I'll add a separate comment about that at the PR level.

Reply:
This is true. I actually could add something here: https://github.com/jrief/django/blob/feae9bc4b4836175089365d0b5d0bfe3b21735f7/django/conf/__init__.py#L236

```python
self.EMAIL_PROVIDER["default"]["BACKEND"] = self.EMAIL_BACKEND
```

The problem then however is how to handle configurations which define the old and the new settings?

Since configurations are handled by the user of a library, for them it would be easy to rewrite them before deploying a new major version of Django.